rpcserver.go

```diff
@@ -1062,12 +1062,127 @@ func (r *rpcServer) VerifyProof(ctx context.Context,
 	)
 	valid := err == nil
 
-	// TODO(roasbeef): also show additional final resting anchor
-	// information, etc?
+	decodedProof, err := r.marshalProofFile(ctx, proofFile, 0, false)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal proof: %w", err)
+	}
+
+	return &taprpc.VerifyProofResponse{
+		Valid:        valid,
+		DecodedProof: decodedProof,
+	}, nil
+}
+
+// DecodeProof attempts to decode a given proof file that claims to be anchored
+// at the specified genesis point.
+func (r *rpcServer) DecodeProof(ctx context.Context,
+	in *taprpc.DecodeProofRequest) (*taprpc.DecodeProofResponse, error) {
+
+	if len(in.RawProof) == 0 {
+		return nil, fmt.Errorf("proof file must be specified")
+	}
+
+	var proofFile proof.File
+	if err := proofFile.Decode(bytes.NewReader(in.RawProof)); err != nil {
+		return nil, fmt.Errorf("unable to decode proof file: %w", err)
+	}
+
+	decodedProof, err := r.marshalProofFile(
+		ctx,
+		proofFile,
+		in.ProofAtDepth,
+		in.WithPrevWitnesses,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal proof: %w", err)
+	}
+
+	return &taprpc.DecodeProofResponse{
+		DecodedProof: decodedProof,
+	}, nil
+}
+
+// marshalProofFile turns a proof file into an RPC DecodedProof.
+func (r *rpcServer) marshalProofFile(ctx context.Context, proofFile proof.File,
+	index int64, withPrevWitnesses bool) (*taprpc.DecodedProof, error) {
+
+	var (
+		decodedProof *proof.Proof
+		err          error
+	)
+
+	// Default to latest proof
+	index = int64(proofFile.NumProofs()) - 1 - index
```

Comment:
What would be a superior UX?
```
[tapcli] unable to verify file: rpc error: code = Unknown desc = unable to marshal proof: invalid index 4294967295
```

The above is the current error message returned. I modified it so it returns the index you actually input, and I think this is most helpful to the user instead of making any assumptions about the proper index:
```
[tapcli] unable to verify file: rpc error: code = Unknown desc = unable to marshal proof: invalid index 2 is greater than latest proof index of 1
```

Reply:
I think it's okay the way it is, but maybe we should make the `proof_at_depth` field `uint32` to avoid it being negative? And I don't think files with more than 2 billion proofs will ever exist, so 32bit should be enough.